### PR TITLE
Save ONNX model in file

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Must be merged after https://github.com/microsoft/onnxruntime/pull/4668

This PR:
* Fixes ORTTrainer initialization when options=None
* Implements save_as_onnx API
* Extend unit test to save, load and compare ONNX models

### CODE REVIEW

File: orttraining/orttraining/python/training/orttrainer.py [hunk 1]
```diff
@@ -447,3 +423,70 @@ def _create_ort_training_session(self):
...
+            self._onnx_model = postprocess.run_postprocess(self._onnx_model)
+
+        # Perform user-specified post-processing
+        if self.options._internal_use.extra_postprocess:
+            self.options._internal_use.extra_postprocess(self._onnx_model)
+
+        # Create training session used by train_step
+        self._create_ort_training_session()
+
+    def _prepare_model_input(self, inputs_desc, lr, loss_scale, *inputs, **kwargs):
```
Comment (liqunfu) [inline orttraining/orttraining/python/training/orttrainer.py:459]: I assume this and above 2 functions are just relocated. There is no change to it.

Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/orttrainer.py:459]: Yes. Just sorted all methods by name

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py [hunk 2]
```diff
@@ -524,3 +524,20 @@ def testInstantiateORTTrainer(step_fn):
...
+    os.remove(file_name)
+
+    # Create a new trainer from persisted ONNX model and compare with original ONNX model
+    trainer_from_onnx = orttrainer.ORTTrainer(reload_onnx_model, model_desc, optim_config)
+    trainer_from_onnx.train_step(data, targets)
+    assert trainer_from_onnx._onnx_model is not None
+    assert (id(trainer_from_onnx._onnx_model) != id(trainer._onnx_model))
+    assert (trainer_from_onnx._onnx_model == trainer._onnx_model)
+    assert (trainer_from_onnx._onnx_model.graph == trainer._onnx_model.graph)
+    assert (onnx.helper.printable_graph(trainer_from_onnx._onnx_model.graph) == onnx.helper.printable_graph(trainer._onnx_model.graph))
```
Comment (liqunfu) [inline orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py:543]: do you want to confirm weighs are indeed updated - if _update_onnx_model_initializers is not called, this test will still pass, right?

File: orttraining/orttraining/python/training/orttrainer.py (hunk 1 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/orttrainer.py:459]: `_update_onnx_model_initializers` is a new method and the other 3 were just relocated

File: orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py (hunk 2 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py:543]: You are right, we are not comparing weights here, just the structure of the graph. We plan on adding such tests for Checkpointing soon

**Pull request merged**